rubicon_ml/cli.py

```diff
@@ -78,7 +82,75 @@ def ui(root_dir, page_size, project_name, host, port, debug, storage_options):
     dashboard.serve(run_server_kwargs=run_server_kwargs)
 
 
-# CLI groups
+@cli.command(
+    context_settings=dict(
+        ignore_unknown_options=True,
+    )
+)
+@click.option(
+    "--root-dir",
+    type=click.STRING,
+    default=os.environ.get("RUBICON_ROOT_DIR"),
+    help="The absolute path to the top level folder holding the rubicon-ml project.",
+    required=False,
+)
+@click.option(
+    "--project-name",
+    default=os.environ.get("RUBICON_PROJECT_NAME"),
+    type=click.STRING,
+    help="Name of rubicon project to retrieve experiments from",
+    required=False,
+)
+@click.option(
+    "--color",
+    "-c",
+    type=click.STRING,
+    help="Color of query output",
+    required=False,
+)
+@click.option(
+    "--bf",
+    is_flag=True,
+    show_default=True,
+    default=True,
+    required=False,
+    help="Flag that switches pretty print output format to basic format",
+)
+@click.argument(
+    "query",
+    nargs=1,
+    required=True,
+)
+def search(root_dir, project_name, color, bf, query):
+    """
+    This function provides capability to query  rubicon experiments from the command line without using python
+    directly and output the JSON in the command line.
+    """
 
+    if project_name is None or root_dir is None:
+        click.secho("No --root-dir or --project-name provided. Exiting...", fg="red")
+        sys.exit(1)
+
+    rubicon = Rubicon(persistence="filesystem", root_dir=root_dir)
+    try:
+        project = rubicon.get_project(name=project_name)
+    except Exception as e:
+        click.secho(e, fg="red")
+        sys.exit(1)
+
+    rb_json = RubiconJSON(projects=project)
+    res = rb_json.search(query)
+    if not isinstance(res, list):
+        res = [res]
+    matches = [r.value for r in res]
+    if "color" in click.get_current_context().params:
+        fg_color = color
+
+    result = matches if bf else pprint.pformat(matches, indent=1)
```

Comment:
what is `bf`?

Reply:
that and the color option were for formatting the output. I replaced `bf` with `pp` for pretty-print, which is what `bf` was actually toggling. I also removed the color output cause we really don't need to support that